core/src/main/java/org/bitcoinj/wallet/DefaultKeyChainFactory.java

```diff
@@ -48,11 +48,14 @@ public DeterministicKeyChain makeKeyChain(Protos.Key key, Protos.Key firstSubKey
     @Override
     public DeterministicKeyChain makeWatchingKeyChain(Protos.Key key, Protos.Key firstSubKey, DeterministicKey accountKey,
                                                       boolean isFollowingKey, boolean isMarried) throws UnreadableWalletException {
+        if (accountKey.getPath().size() != DeterministicKeyChain.ACCOUNT_ZERO_PATH.size())
```

Comment:
This check was added back, with the change that it checks for the size of the account path (being 1) rather than the value being 0H.

Otherwise, if path with longer length would cause different exception in `DeterministicKeyChain(DeterministicKey watchingKey)`